intercom_flutter/android/src/main/java/io/maido/intercom/PushInterceptReceiver.kt

```diff
@@ -0,0 +1,36 @@
+package io.maido.intercom
+
+import android.app.Application
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.util.Log
+import com.google.firebase.messaging.RemoteMessage
+import io.intercom.android.sdk.push.IntercomPushClient
+
+class PushInterceptReceiver : BroadcastReceiver() {
+
+    private val intercomPushClient = IntercomPushClient()
+
+    override fun onReceive(context: Context?, intent: Intent?) {
+        if (context == null) return
+
+        val application = context.applicationContext as Application
+
+        val dataBundle = intent?.extras ?: return
+
+        val remoteMessage = RemoteMessage(dataBundle)
+        val message = remoteMessage.data
+
+        if (intercomPushClient.isIntercomPush(message)) {
+            Log.d(TAG, "Intercom message received")
+            intercomPushClient.handlePush(application, message)
+        } else {
+            Log.d(TAG, "Push message received, not for Intercom")
```

Comment:
So if there is a push notification from Intercom then the user will see 2 notifications.
- one because of here (handled by the Intercom)
- Second because of Firebase messaging handler.

right?

Reply:
not sure what you mean by "see 2 notifications", but he will receive it through 2 handlers, one is the Firebase handler, and the other is this one, we can still use `Intercom.isIntercomPush()` to ignore processing Intercom notifications via Firebase handler if required...